### PR TITLE
Add buffered lookahead for Jackson

### PR DESCRIPTION
**Pull request by luneo7**
### Description
Ports buffered lookahead for Jackson from elasticsearch-java https://github.com/elastic/elasticsearch-java/pull/489
Improving performance hit caused by previous implementation.

### Issues Resolved
Fixes https://github.com/opensearch-project/opensearch-java/issues/337

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@luneo7 could you please fix DCO (https://github.com/opensearch-project/opensearch-java/blob/main/CONTRIBUTING.md#developer-certificate-of-origin) and add CHANGELOG entry, thank you

**Review by dblock — DISMISSED** [review]

**Review by dblock — APPROVED** [review]

**Comment by luneo7** [comment]
@reta can you do a review of this PR? Thanks =]

**Comment by reta** [comment]
> @reta can you do a review of this PR? Thanks =]

@luneo7  my apologies, I am on vacation    at the moment, should be back at the end of the upcoming week

**Comment by dblock** [comment]
This LGTM. I think we need some USER_GUIDE updates for this?

**Review by reta — DISMISSED** [review]

File: java-client/src/main/java/org/opensearch/client/json/jackson/JacksonJsonpMapper.java [hunk 1]
```diff
@@ -54,18 +54,20 @@ public class JacksonJsonpMapper extends JsonpMapperBase {
...
-        this.objectMapper = objectMapper
-            .configure(SerializationFeature.INDENT_OUTPUT, false)
-            .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        this.objectMapper = objectMapper;
     }
 
     public JacksonJsonpMapper() {
-        this(new ObjectMapper());
+        this(new ObjectMapper()
+            .configure(SerializationFeature.INDENT_OUTPUT, false)
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/json/jackson/JacksonJsonpMapper.java:68]: Curious, is this change related to the backport?

Comment (luneo7) [reply java-client/src/main/java/org/opensearch/client/json/jackson/JacksonJsonpMapper.java:68]: More or less (https://github.com/elastic/elasticsearch-java/blob/main/java-client/src/main/java/co/elastic/clients/json/jackson/JacksonJsonpMapper.java#L56) when back porting the Jackson buffered lookahead change, I also saw that the constructor was different, and saw that they ES improved the constructor so if we pass an existing ObjectMapper it doesn't changes its configuration. For me it is a win, since we use a singleton instance of objectmapper and we had to clone it to construct the opensearch client... since it sets `JsonInclude.Include.NON_NULL` and we use `JsonInclude.Include.NON_ABSENT`... so this change actually makes the constructor with the objectmapper param not change the objectmapper config as it was doing before.

**Comment by luneo7** [comment]
@dblock @reta can you review it again? After I fixed the conflict on the CHANGELOG it lost approvals

**Review by reta — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by luneo7** [comment]
Can it be backported to 2.x? @dblock @reta
